lollipop_jsonschema/jsonschema.py

```diff
@@ -479,33 +483,40 @@ def json_schema(self, schema, definitions=None, mode=None):
         definition_names = {definition.name
                             for definition in itervalues(definitions)}
 
-        counter = SchemaUsageCounter(self._encoders)
+        counter = SchemaUsageCounter(self._encoders, mode=mode)
         counter.json_schema(schema)
         counts = counter.counts
 
-        for schema1, count in iteritems(counts):
+        mode_suffix = '_' + mode if mode else ''
+
+        for schema_key, count in iteritems(counts):
             if count == 1:
                 continue
 
-            if schema1 not in definitions:
-                def_name = _sanitize_name(schema1.name) if schema1.name else 'Type'
+            schema1, mode = schema_key
+            if schema_key not in definitions:
```

Comment:
For a note here, this seems to be a backward incompatible change because the key type is changed.